core/templates/domain/user/user-info.model.ts

```diff
@@ -63,30 +63,30 @@ export class UserInfo {
   static createFromBackendDict(
       data: UserInfoBackendDict): UserInfo {
     return new UserInfo(
-      data.role, data.is_moderator, data.is_admin, data.is_super_admin,
-      data.is_topic_manager, data.can_create_collections,
+      data.roles, data.is_moderator, data.is_curriculum_admin,
+      data.is_super_admin, data.is_topic_manager, data.can_create_collections,
       data.preferred_site_language_code, data.username,
       data.email, data.user_is_logged_in);
   }
   static createDefault(): UserInfo {
     return new UserInfo(
-      'GUEST', false, false, false, false, false, null, null, null, false);
+      ['GUEST'], false, false, false, false, false, null, null, null, false);
```

Comment:
Do we have a constant for this role? Ditto for the roles below.

Reply:
Not in the frontend as of now, but we are planning to move all the roles constants to fronted which will simplify the UserInfo constructor as well. (Going to be @ashutoshc8101 next PR!)

Reply:
Ok, thanks!